src/Admin/SiteHealth.php

```diff
@@ -223,6 +294,126 @@ static function ( $available_service ) {
+		$status      = 'recommended';
+		$label       = __( 'Page caching is not detected', 'amp' );
+
+		$description = esc_html__( 'The AMP plugin performs at its best when page caching is enabled. This is because the additional optimizations performed require additional server processing time, and page caching ensures that responses are served quickly.', 'amp' );
+
+		if ( $caching_status['server_caching'] && ! $caching_status['client_caching'] ) {
+			$badge_color = 'orange';
+			$status      = 'recommended';
+			$label       = __( 'Page caching is detected but client caching headers are missing', 'amp' );
+		} elseif ( $caching_status['server_caching'] && $caching_status['client_caching'] ) {
+			$badge_color = 'green';
+			$status      = 'good';
+			$label       = __( 'Page caching is detected', 'amp' );
+		}
+
+		return [
+			'badge'       => [
+				'label' => $this->get_badge_label(),
+				'color' => $badge_color,
+			],
+			'description' => $description,
+			'test'        => self::TEST_PAGE_CACHING,
+			'status'      => $status,
+			'label'       => esc_html( $label ),
+			'actions'     => sprintf(
+				'<p><a href="%1$s" target="_blank" rel="noopener noreferrer">%2$s <span class="screen-reader-text">%3$s</span><span aria-hidden="true" class="dashicons dashicons-external"></span></a></p>',
+				esc_url( 'https://amp-wp.org/documentation/getting-started/amp-site-setup/page-caching-with-amp-and-wordpress/' ),
+				esc_html__( 'Learn more about page caching', 'amp' ),
+				/* translators: The accessibility text. */
+				esc_html__( '(opens in a new tab)', 'amp' )
+			),
+		];
+	}
+
+	/**
+	 * Check if site has page cache enable or not.
+	 *
+	 * @return array {
+	 *     @type bool $server_caching True if site has page cache. Otherwise False.
+	 *     @type bool $client_caching True if site has page cache and passed appropriate header.
+	 * }
+	 */
+	private function get_page_cache_status() {
+
+		$has_page_cache = [
+			'server_caching' => false,
+			'client_caching' => false,
+		];
+		$response_list  = [];
+		$request_args   = [
+			'headers' => [
+				'X-AMP-Request-Random-Number' => true,
+				'Cache-Control'               => 'max-age=' . ( YEAR_IN_SECONDS + MONTH_IN_SECONDS ),
+			],
+		];
+
+		for ( $i = 1; $i <= 3; $i++ ) {
+			$http_response        = wp_remote_get( home_url(), $request_args );
+			$http_response_header = wp_remote_retrieve_headers( $http_response );
+			$http_response_header = ( ! empty( $http_response_header ) ) ? $http_response_header : [];
+			$http_response_header = ( is_array( $http_response_header ) ) ? $http_response_header : $http_response_header->getAll();
+
+			$response = false;
+
+			// If any of header tag exist. and x-amp-random-number doesn't consider page cache exists.
+			if (
+				! empty( $http_response_header['cache-control'] ) ||
+				! empty( $http_response_header['expires'] ) ||
+				! empty( $http_response_header['last-modified'] ) ||
+				! empty( $http_response_header['etag'] )
```

Comment:
Come to think of it, perhaps `Last-Modified` and `ETag` wouldn't be so key in terms of caching. At least, they would indicate a lesser degree of frontend-caching because they'd require a client to send `If-Modified-Since`/`If-None-Match` headers for a conditional request. Even so, they don't necessarily mean that there is any caching present at all, but rather that the server is able to save the client some bytes from re-sending the same response.

So if we do, then we should be sending them in the subsequent request(s). If the subsequent requests return a `304 Not Modified` response, then that is the necessary positive signal for caching.

Reply:
WordPress has built-in support for conditional requests specifically for sending `Last-Modified` and `ETag` response headers for feed requests: https://github.com/WordPress/wordpress-develop/blob/57ad9f16dff35a879e856566489266472589ea04/src/wp-includes/class-wp.php#L435-L516